### PR TITLE
feat(pci.data-processing): submitJob - improve sizing and add prices details

### PR DESCRIPTION
**Pull request by jperchoc**
DATATR-15

Signed-off-by: Jonathan Perchoc <jonathan.perchoc@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/datatr-14-jobs-new-layout`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #DATATR-15
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Improve sizing step
Add prices computation

## Related

```
DATATR-2 <-- DATATR-14 <-- DATATR-15 <-- DATATR-16
```
[DATATR-2](https://github.com/ovh/manager/tree/feat/datatr-2-dataprocessing-new-job)
[DATATR-14](https://github.com/ovh/manager/pull/7047)
[DATATR-15](https://github.com/ovh/manager/pull/7067)
[DATATR-16](https://github.com/ovh/manager/pull/7069)

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/price-details/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  "data_processing_price_details_memory": "Mémoire",
+  "data_processing_price_details_estimated_price": "Prix estimé",
+  "data_processing_price_details_driver": "Driver : ",
+  "data_processing_price_details_workers": "Executors : ",
+  "data_processing_price_details_memory_gib": "{{memory  | number:2}} GiB",
+  "data_processing_price_details_memory_overhead": "overhead",
+  "data_processing_price_details_price_per_core": "Prix par core : ",
+  "data_processing_price_details_price_per_gib": "Prix par GiB : ",
+  "data_processing_price_details_total": "Total : ",
+  "data_processing_price_details_unit_minute": "minute",
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/price-details/translations/Messages_fr_FR.json:14]: **suggestion:** Can be maintained as constant.

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/price-details/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "data_processing_price_details_unit_hour": "heure",
+  "data_processing_price_details_unit_day": "jour",
+  "data_processing_price_details_unit_minute_plural": "minutes",
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/price-details/translations/Messages_fr_FR.json:17]: **suggestion:** Can be maintained as constant.

Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/price-details/translations/Messages_fr_FR.json:17]: Hi @anooparveti, I think that those need to be translated, I am not certain we can say "minutes" in Espagnol or in Polish languages

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/price-details/translations/Messages_fr_FR.json (hunk 1 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/price-details/translations/Messages_fr_FR.json:14]: Same as here: https://github.com/ovh/manager/pull/7067#discussion_r905058601

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/price-details/price-details.html [hunk 3]
```diff
@@ -0,0 +1,235 @@
+<div class="mb-3">
+    <h5 data-translate="data_processing_price_details_estimation"></h5>
+    <p data-translate="data_processing_price_details_description"></p>
+    <div class="mb-2 d-flex align-items-center">
+        <span
+            class="mr-2"
+            data-translate="data_processing_price_details_estimation_select_unit"
+        >
+        </span>
+        <label class="oui-select oui-select_inline ml-2">
```
[the remark this answers is not in the document]
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/price-details/price-details.html]: Added here: https://github.com/ovh/manager/pull/7067/commits/8c4a71be77f20cd3e7f33af1528fc24812f0c0ba

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html [hunk 4]
```diff
@@ -1,219 +1,237 @@
...
+                            <oui-numeric
+                                name="driver-memory-overhead"
+                                data-model="$ctrl.state.driverMemoryOverheadMb"
+                                data-min="$ctrl.jobParameters.driver_memory_overhead.validator.min"
+                                data-max="$ctrl.maxOverheadMemoryDriverGb"
+                                data-disabled="$ctrl.state.advancedSizingDriverMemOverheadAuto"
+                            >
+                            </oui-numeric>
+                        </div>
+                        <span class="sizing__advanced-units">MiB</span>
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html:129]: **question:** Dont we need translation for `MiB`?

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html [hunk 5]
```diff
@@ -1,219 +1,237 @@
...
+                            <oui-numeric
+                                name="worker-memory"
+                                data-model="$ctrl.state.workerMemoryGb"
+                                data-min="$ctrl.jobParameters.executor_memory.validator.min / 1024"
+                                data-max="$ctrl.maxMemoryWorkerGb"
+                                data-on-change="$ctrl.onChangeWorkerMemoryHandler(modelValue)"
+                            >
+                            </oui-numeric>
+                        </div>
+                        <span class="sizing__advanced-units">GiB</span>
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html:177]: Same as https://github.com/ovh/manager/pull/7067/files#r910926052

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html [hunk 6]
```diff
@@ -1,219 +1,237 @@
...
+                            <oui-numeric
+                                name="worker-memory-overhead"
+                                data-model="$ctrl.state.workerMemoryOverheadMb"
+                                data-min="$ctrl.jobParameters.executor_memory_overhead.validator.min"
+                                data-max="$ctrl.maxOverheadMemoryWorkerGb"
+                                data-disabled="$ctrl.state.advancedSizingWorkerMemOverheadAuto"
+                            >
+                            </oui-numeric>
+                        </div>
+                        <span class="sizing__advanced-units">MiB</span>
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html:194]: Same as https://github.com/ovh/manager/pull/7067/files#r910926052

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html [hunk 7]
```diff
@@ -1,219 +1,237 @@
...
+                            class="sizing__advanced-units-checkbox pl-0"
+                            name="worker-memory-overhead-auto"
+                            data-model="$ctrl.state.advancedSizingWorkerMemOverheadAuto"
+                            data-on-change="$ctrl.onChangeAdvancedSizingWorkerMemOverheadHandler(modelValue)"
+                        >
+                            <span
+                                >{{
+                                ::'data_processing_submit_job_stepper_spark_memory_overhead_autocompute_label'
+                                | translate}}</span
+                            >
```
[the remark this answers is not in the document]
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html]: Done here: https://github.com/ovh/manager/pull/7067/commits/85a5cac61eebc3109c240870e7b9858a3345f8da

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html (hunk 6 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html]: https://github.com/ovh/manager/pull/7067/commits/85a5cac61eebc3109c240870e7b9858a3345f8da

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html (hunk 5 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html]: https://github.com/ovh/manager/pull/7067/commits/85a5cac61eebc3109c240870e7b9858a3345f8da

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html [hunk 8]
```diff
@@ -1,219 +1,237 @@
...
+                            class="sizing__advanced-units-checkbox pl-0"
+                            name="driver-memory-overhead-auto"
+                            data-model="$ctrl.state.advancedSizingDriverMemOverheadAuto"
+                            data-on-change="$ctrl.onChangeAdvancedSizingDriverMemOverheadHandler(modelValue)"
+                        >
+                            <span
+                                >{{
+                                ::'data_processing_submit_job_stepper_spark_memory_overhead_autocompute_label'
+                                | translate}}</span
+                            >
```
[the remark this answers is not in the document]
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html]: https://github.com/ovh/manager/pull/7067/commits/85a5cac61eebc3109c240870e7b9858a3345f8da

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html (hunk 4 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html]: https://github.com/ovh/manager/pull/7067/commits/85a5cac61eebc3109c240870e7b9858a3345f8da

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html [hunk 9]
```diff
@@ -1,219 +1,237 @@
...
+                            <oui-numeric
+                                name="driver-memory"
+                                data-model="$ctrl.state.driverMemoryGb"
+                                data-min="$ctrl.jobParameters.driver_memory.validator.min / 1024"
+                                data-max="$ctrl.maxMemoryDriverGb"
+                                data-on-change="$ctrl.onChangeDriverMemoryHandler(modelValue)"
+                            >
+                            </oui-numeric>
+                        </div>
+                        <span class="sizing__advanced-units">GiB</span>
```
[the remark this answers is not in the document]
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-sizing/spark-sizing.html]: https://github.com/ovh/manager/pull/7067/commits/85a5cac61eebc3109c240870e7b9858a3345f8da

**Review by anooparveti — APPROVED** [review]

**Review by antleblanc — CHANGES_REQUESTED** [review]
@jperchoc Could you please resolve existing conflicts?
Thank you!

**Pull request merged**